### PR TITLE
docs(guides): update SplitChunksPlugin defaults

### PR DESCRIPTION
**Pull request by rohrlaf**
This PR will update and align default values for `SplitChunksPlugin`.

* set `maxAsyncRequests` and `maxInitialRequests` to `30` (see https://github.com/webpack/webpack.js.org/pull/3819 https://github.com/webpack/webpack.js.org/issues/3804)
* set `chunks` to `all` (see [webpack/SplitChunksPlugin.js#L540](https://github.com/webpack/webpack/blob/master/lib/optimize/SplitChunksPlugin.js#L540))
* align default values with [Writer's Guide](https://webpack.js.org/contribute/writers-guide/#configuration-defaults-and-types)

### CODE REVIEW

File: src/content/plugins/split-chunks-plugin.md [hunk 1]
```diff
@@ -145,38 +146,38 @@ Maximum number of parallel requests at an entry point.
...
 
+#### `splitChunks.cacheGroups.{cacheGroup}.enforceSizeThreshold`
+
 Size threshold at which splitting is enforced and other restrictions (minRemainingSize, maxAsyncRequests, maxInitialRequests) are ignored.
 
 
 ### `splitChunks.minRemainingSize`
 
-#### `splitChunks.cacheGroups.{cacheGroup}.minRemainingSize`
+`number = 0`
```
Comment (EugeneHlushko) [inline src/content/plugins/split-chunks-plugin.md:170]: Moving it here means you have to provide a separate type mention after `#### splitChunks.cacheGroups.{cacheGroup}.minRemainingSize`

Comment (rohrlaf) [reply src/content/plugins/split-chunks-plugin.md:170]: Okay, so I simply should keep the position below `{cacheGroup}.minRemainingSize`?

Comment (rohrlaf) [reply src/content/plugins/split-chunks-plugin.md]: Done in 1958bd5

**Comment by rohrlaf** [comment]
@rishabh3112 Just found a mistake I made for `minSize`, fixed it with 08c70f8 after your approval.

**Comment by EugeneHlushko** [comment]
Thanks!

**Comment by wojtekmaj** [comment]
Not sure about the `chunks` bit. I use webpack 5.0.0-beta.22 and when I added
```
    splitChunks: {
      chunks: 'all',
    },
```

suddenly my entrypoint was split into two files, and in general my output directory was significantly different.

**Comment by rohrlaf** [comment]
> Not sure about the `chunks` bit. I use webpack 5.0.0-beta.22 and when I added
> 
> ```
>     splitChunks: {
>       chunks: 'all',
>     },
> ```
> 
> suddenly my entrypoint was split into two files, and in general my output directory was significantly different.
> 
> Furthermore, [Webpack 5 migration guide](https://github.com/webpack/webpack.js.org/blob/master/src/content/migrate/5.md?rgh-link-date=2020-07-31T12%3A54%3A30Z) states
> 
> > It's recommended to use either the defaults or `optimization.splitChunks: { chunks: 'all' }`.
> 
> suggesting that `chunks`'s default is not `"all"`.

Could be a mistake from my side. I consulted [webpack/SplitChunksPlugin.js#L540](https://github.com/webpack/webpack/blob/master/lib/optimize/SplitChunksPlugin.js#L540) for the default settings as stated in the description. So either there are unintentional side-effects in your configuration or this code snippet does not have the effect it promises. But also, I am not to familiar with Webpack internals and only saw the inconsistencies in the Webpack docs that I wanted to fix.

```js
// SplitChunksPlugin.js
/** @type {SplitChunksOptions} */
this.options = {
  chunksFilter: normalizeChunksFilter(options.chunks || "all"),
  // ...
};
```

@rishabh3112 @EugeneHlushko How should I proceed with this?

**Pull request merged**